expression/builtin_encryption.go

```diff
@@ -494,12 +494,17 @@ type sha2FunctionClass struct {
 }
 
 func (c *sha2FunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinSHA2Sig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpInt)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	types.SetBinChsClnFlag(bf.tp)
```

Comment:
1. result's charset and collation should be utf8, you can just remove line:501 to fix it
2. set `bf.tp.Flen`